flink-cdc-common/src/main/java/com/ververica/cdc/common/event/DataRecord.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.common.event;
+
+/** Class {@code DataRecord} describes the data of changed record in the external system. */
+public interface DataRecord {
+
+    /** Returns the number of fields in this row. */
+    int getArity();
+
+    // ------------------------------------------------------------------------------------------
+    // Read-only accessor methods
+    // ------------------------------------------------------------------------------------------
+
+    /** Returns true if the field is null at the given position. */
+    boolean isNullAt(int pos);
+
+    /** Returns the boolean value at the given position. */
+    boolean getBoolean(int pos);
+
+    /** Returns the byte value at the given position. */
+    byte getByte(int pos);
+
+    /** Returns the short value at the given position. */
+    short getShort(int pos);
+
+    /** Returns the integer value at the given position. */
+    int getInt(int pos);
+
+    /** Returns the long value at the given position. */
+    long getLong(int pos);
+
+    /** Returns the float value at the given position. */
+    float getFloat(int pos);
+
+    /** Returns the double value at the given position. */
+    double getDouble(int pos);
+
+    /** Returns the string value at the given position. */
+    String getString(int pos);
+
+    /** Returns the binary value at the given position. */
+    byte[] getBinary(int pos);
+
+    // TODO: add more methods for other types
```

Comment:
We can support all simple type that Flink SQL supports

Reply:
There are some complex data types remains.  
We need to follow the type structure of Flink, or do we need to wait for the customized type system?
- DecimalData
- TimestampData
- RawValueData
- ArrayData
- MapData
- RowData

Reply:
Yes, good point ,we need more type structure for following methods